extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -534,17 +534,23 @@ export class AmpSelector extends AMP.BaseElement {
     }
 
     event.preventDefault();
-
     // Make currently selected option unfocusable
     this.elements_[this.focusedIndex_].tabIndex = -1;
 
-    // Change the focus to the next element in the specified direction.
-    // The selection should loop around if the user attempts to go one
-    // past the beginning or end.
-    this.focusedIndex_ = (this.focusedIndex_ + dir) % this.elements_.length;
-    if (this.focusedIndex_ < 0) {
-      this.focusedIndex_ = this.focusedIndex_ + this.elements_.length;
-    }
+    const originalIndex = this.focusedIndex_;
+    do {
+      // Change the focus to the next element in the specified direction.
+      // The selection should loop around if the user attempts to go one
+      // past the beginning or end.
+      this.focusedIndex_ = (this.focusedIndex_ + dir) % this.elements_.length;
+      if (this.focusedIndex_ < 0) {
+        this.focusedIndex_ = this.focusedIndex_ + this.elements_.length;
+      }
+
+      if (!isElementHidden(this.elements_[this.focusedIndex_])) {
+        break;
+      }
+    } while (this.focusedIndex_ != originalIndex);
```

Comment:
What happens here if the original index is the only non-hidden index? Seems like focus goes to a hidden element.

Would suggest not updating `this.focusedIndex_` until just before the break.

Reply:
I believe it would go back to the original element if it's the only non-disabled one